applications/crossbar/priv/couchdb/account/devices.json

```diff
@@ -20,6 +20,24 @@
                 "}"
             ]
         },
+        "listing_by_external_cidnum": {
+            "map": [
+                "function (doc) {",
+                "  if (doc.pvt_type != 'device' || doc.pvt_deleted) return;",
+                "  if(doc.caller_id.external.number)",
+                "    emit(doc.caller_id.external.number, {'id': doc._id})",
```

Comment:
You get `_id` in the view results by default so no need to emit it. You get "key", "value", and "id" (and "doc" if you include docs) in each view result.